some docs fixes (replacing old links with new one ) (#7762)
sunpy/sunpy: docs/dev_guide/contents/newcomers.rst

The instructions in this following section are based upon three resources:

* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`__
* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`__

**We strongly recommend that you read these links.**
These links are more in-depth than this guide but you will need to replace ``astropy`` with ``sunpy``.

> In order to start coding you will need a local Python environment and we would recommend using Anaconda or `miniconda`_ (shortened to conda from here on).

Reviewer: ```suggestion
In order to start coding you will need a local Python environment and we would recommend using `miniforge`_ (shortened to conda from here on).
```

The instructions in this following section are based upon three resources:

> * `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`_
> * `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_

Reviewer: ```suggestion
* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`__
* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`__
```

The instructions in this following section are based upon three resources:

> * `Astropy Dev Workflow <https://kbarbary-astropy.readthedocs.io/en/latest/development/workflow/development_workflow.html>`_

Reviewer 2: ```suggestion
```

Reviewer 2: This page doesn't exist anymore.  We definitely shouldn't link to someone else's rendering of super-old Astropy docs.

> * `Astropy Dev environment <https://docs.astropy.org/en/latest/development/quickstart.html>`_

Reviewer 2: ```suggestion
* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`_
```

> * `Astropy Pull Request Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_

Reviewer 2: ```suggestion
* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_
```

You can change ``my_fix`` to anything you prefer.
If you get stuck or want help, just `ask here`_!

> .. _miniconda: https://conda.io/en/latest/miniconda.html

Reviewer: ```suggestion
.. _miniforge: hhttps://github.com/conda-forge/miniforge?tab=readme-ov-file#download
```
